Firmware/MotorControl/axis.cpp

```diff
@@ -20,13 +30,23 @@ Axis::Axis(int axis_num,
       encoder_(encoder),
       sensorless_estimator_(sensorless_estimator),
       controller_(controller),
+      fet_thermistor_(fet_thermistor),
+      motor_thermistor_(motor_thermistor),
       motor_(motor),
-      trap_(trap)
+      trap_(trap),
+      current_limiters_(make_array(
+          static_cast<CurrentLimiter*>(&fet_thermistor),
+          static_cast<CurrentLimiter*>(&motor_thermistor))),
```

Comment:
as far as I can see you can do the same without `make_array`:
`current_limiters_{static_cast<CurrentLimiter*>(&fet_thermistor), static_cast<CurrentLimiter*>(&motor_thermistor)}`

Reply:
@samuelsadok yes you can but you lose some safety. For instance the following will compile:

```
class Foo {
    Foo() : content({"test", "test2"}) {}
  
    std::array<std::string, 3> content;
};
```

The `make_array` helpers ensures that you are forced to fill in every element.

Reply:
Ah I see. I assume in this example the default constructor would be used for the third string. Then perhaps in our case it makes sense to check for `nullptr` in the loops that iterate over these arrays. Either that or `make_array` can be moved to "utils.hpp".